torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -941,7 +941,7 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       getCvarInt(TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC, 60 * 8 /*8 Mins*/);
   waitTimeoutDumpInMilSec_ =
       getCvarInt(TORCH_NCCL_WAIT_TIMEOUT_DUMP_MILSEC, 60 * 1000 /*60 Sec*/);
-  coordCheckIntervalMilSec_ = getCvarInt(TORCH_NCCL_COORD_CHECK_MILSEC, 1000);
+  coordCheckIntervalMilSec_ = getCvarInt(TORCH_NCCL_COORD_CHECK_MILSEC, 300);
```

Comment:
Curious: why would the checking frequency matter?

Reply:
because while one rank is sleeping, other ranks already finished the dump the FR and exit early. And the trainer just killed the training process.

Reply:
so basically I noticed that for some missing dump cases, we don't even print the log where we way that we observed the dump signals from TCPStore, while other ranks do report that, this indicates that this rank might be just sleeping? If we think this is too aggressive, I can change it to larger value?

Reply:
> Which thread finishes dumping the FR -- Heartbeat or watchdog?
Right now, Heartbeat thread.

> Which thread throws an exception which causes torchrun to broadcast killing to other ranks?
Watchdog thread

> Which thread keeps the TCPStore server alive so that other ranks can sense the dump request?
For this case, as long as rank 0 is alive, TCPStore server is alive.

I think you confused about thread coordination and rank coordination. Here I am talking about rank coordination, basically other ranks exit/throw exception early so we are seeing logs message below:

> NCCL INFO [Service thread] Connection closed by localRank 4
> NCCL INFO [Service thread] Connection closed by localRank 1

Reply:
So basically this is beyond thread level, so it does not matter which thread is doing the dump. That thread needs to get dump signal from TCPStore because the rank itself is not aware of nccl watchdog timeout.